### PR TITLE
fix(bridge): do not allow dot in bridge name

### PR DESCRIPTION
**Pull request by zmstone**
Do not allow dot in bridge name or connector name.

bridge and connector names are used as config keys in cluster.hocon.
dots which will turn the key into path.

this is a follow up fix of bridge-v2 feature which is not released yet.
so there is no compatibility issue.

### <samp>🤖 Generated by Copilot at 039976f</samp>

This pull request refactors and improves the validation of resource IDs, types, and names for bridges and connectors, using common functions from the `emqx_resource` module. It also adds and renames dummy connector modules for testing purposes.

### CODE REVIEW

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_bridge/src/emqx_bridge_v2_api.erl [hunk 1]
```diff
@@ -742,11 +742,24 @@ update_bridge(BridgeType, BridgeName, Conf) ->
...
 create_or_update_bridge(BridgeType, BridgeName, Conf, HttpStatusCode) ->
-    case emqx_bridge_v2:create(BridgeType, BridgeName, Conf) of
-        {ok, _} ->
-            lookup_from_all_nodes(BridgeType, BridgeName, HttpStatusCode);
-        {error, Reason} when is_map(Reason) ->
-            ?BAD_REQUEST(map_to_json(redact(Reason)))
+    Check =
+        try
+            ok = emqx_resource:validate_type(BridgeType),
+            ok = emqx_resource:validate_name(BridgeName)
```
[the remark this answers is not in the document]
Comment (zmstone) [reply apps/emqx_bridge/src/emqx_bridge_v2_api.erl:748]: as it turns out, it's already parsed in the wrapping macro.

[the remark this answers is not in the document]
Comment (zmstone) [reply apps/emqx_bridge/src/emqx_bridge_v2_api.erl:748]: i mean, wrapping schema.
but we do need to add validation to name.

File: apps/emqx_bridge/src/emqx_bridge_resource.erl [hunk 2]
```diff
@@ -102,21 +102,15 @@ bridge_id(BridgeType, BridgeName) ->
     <<Type/binary, ":", Name/binary>>.
 
 parse_bridge_id(BridgeId) ->
-    parse_bridge_id(BridgeId, #{atom_name => true}).
+    parse_bridge_id(bin(BridgeId), #{atom_name => true}).
 
--spec parse_bridge_id(list() | binary() | atom(), #{atom_name => boolean()}) ->
+-spec parse_bridge_id(binary() | atom(), #{atom_name => boolean()}) ->
     {atom(), atom() | binary()}.
+parse_bridge_id(<<"birdge:", ID/binary>>, Opts) ->
```
Comment (thalesmg) [inline apps/emqx_bridge/src/emqx_bridge_resource.erl:109]:
```suggestion
parse_bridge_id(<<"bridge:", ID/binary>>, Opts) ->
```

File: apps/emqx_resource/src/emqx_resource.erl [hunk 3]
```diff
@@ -776,3 +783,79 @@ clean_allocated_resources(ResourceId, ResourceMod) ->
...
+%% @doc Validate if name is valid for bridge.
+%% Throws and JSON-map error if invalid.
+-spec validate_name(binary()) -> ok.
+validate_name(Name) ->
+    _ = validate_name(Name, #{atom_name => false}),
+    ok.
+
+validate_name(<<>>, _Opts) ->
+    invalid_data("name cannot be empty string");
+validate_name(Name, _Opts) when size(Name) >= 255 ->
```
Comment (thalesmg) [inline apps/emqx_resource/src/emqx_resource.erl:834]:
```suggestion
validate_name(Name, _Opts) when size(Name) > 255 ->
```

File: apps/emqx_resource/src/emqx_resource.erl [hunk 4]
```diff
@@ -776,3 +783,79 @@ clean_allocated_resources(ResourceId, ResourceMod) ->
...
+
+-spec invalid_data(binary()) -> no_return().
+invalid_data(Reason) -> throw(#{kind => validation_error, reason => Reason}).
+
+is_id_char(C) when C >= $0 andalso C =< $9 -> true;
+is_id_char(C) when C >= $a andalso C =< $z -> true;
+is_id_char(C) when C >= $A andalso C =< $Z -> true;
+is_id_char($_) -> true;
+is_id_char($-) -> true;
+is_id_char(_) -> false.
```
Comment (thalesmg) [inline apps/emqx_resource/src/emqx_resource.erl:861]: Q: is there a reason we didn't use a regex for this?

```suggestion
    case re:run(Name0, <<"^[-0-9a-zA-Z_]+$">>, [{capture, none}]) of
        match ->
            case maps:get(atom_name, Opts, true) of
                % NOTE
                % Rule may be created before bridge, thus not `list_to_existing_atom/1`,
                % also it is infrequent user input anyway.
                true -> binary_to_atom(Name0, utf8);
                false -> Name0
            end;
        nomatch ->
            invalid_data(
                <<"only 0-9a-zA-Z_- is allowed in resource name, got: ", Name0/binary>>
            )
    end.

-spec invalid_data(binary()) -> no_return().
invalid_data(Reason) -> throw(#{kind => validation_error, reason => Reason}).
```

**Review by thalesmg — APPROVED** [review]

**Pull request merged**